### PR TITLE
Bakers reward period

### PR DESCRIPTION
**Pull request by EmilLa1**
## Purpose

Solve https://github.com/Concordium/concordium-rust-sdk/issues/107

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Review by td202 — APPROVED** [review]
Looks good.

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,8 @@
 - Add a `Cis4Contract` for interacting with Cis4 contracts.
 - Add a new `web3id` module that contains types and functionality for
   construcing Web3ID credentials and verifying Web3ID proofs.
+- Add support for `GetBakersRewardPeriod` endpoint (only available when querying a node
+  with at least version 6.1)
```
Comment (abizjak) [inline CHANGELOG.md:22]: Move up to unreleased changes section.

File: src/types/mod.rs [hunk 2]
```diff
@@ -2238,6 +2238,32 @@ pub struct NodeInfo {
     pub details:      NodeDetails,
 }
 
+/// Information of a baker for a certain reward period.
+#[derive(Debug)]
+pub struct BakerRewardPeriodInfo {
+    /// Metadata for the baker.
```
Comment (abizjak) [inline src/types/mod.rs:2244]: You are introducing a new term here "metadata" that does not really explain anything. Rephrase please.

File: src/types/mod.rs [hunk 3]
```diff
@@ -2238,6 +2238,32 @@ pub struct NodeInfo {
...
+    pub baker:             BakerInfo,
+    /// The stake of the baker that the
+    /// consensus protocol uses to determine lottery weight.
+    /// I.e. this is the stake after applying leverage bound and caps.
```
Comment (abizjak) [inline src/types/mod.rs:2248]: Don't start sentences with `i.e.`. If you just omit it here the meaning stays the same.

File: src/types/mod.rs (hunk 2 above)
Comment (abizjak) [reply src/types/mod.rs:2244]: In the proto file there is a better line
```
/// The baker id and public keys for the baker.
```

File: src/v2/mod.rs [hunk 4]
```diff
@@ -2443,6 +2443,28 @@ impl Client {
         }
         last_found.ok_or(QueryError::NotFound)
     }
+
+    /// Get the bakers of a reward period.
```
Comment (abizjak) [inline src/v2/mod.rs:2447]: Please include the details that are present in the service description already.

File: src/v2/mod.rs [hunk 5]
```diff
@@ -2443,6 +2443,31 @@ impl Client {
         }
         last_found.ok_or(QueryError::NotFound)
     }
+
+    /// Get all bakers in the reward period of a block.
+    /// This endpoint is only supported for protocol version 6 and onwards.
+    /// If the protocol does not support the endpoint then an
+    /// 'IllegalArgument' error is returned.
```
Comment (abizjak) [inline src/v2/mod.rs:2450]:
```suggestion
    /// [`IllegalArgument`](tonic::Code::InvalidArgument) is returned.
```

Let's not use haddock markup in Rust, it does not work.

**Review by abizjak — APPROVED** [review]

**Pull request merged**